src/pybind/mgr/orchestrator_cli/module.py

```diff
@@ -82,27 +82,24 @@ def _wait(self, completions):
 
         Waits for writes to be *persistent* but not *effective*.
         """
-        done = False
-
-        while done is False:
-            done = self._oremote("wait", completions) == []
-
-            if not done:
-                any_nonpersistent = False
-                for c in completions:
-                    if c.is_read:
-                        if not c.is_complete:
-                            any_nonpersistent = True
-                            break
-                    else:
-                        if not c.is_persistent:
-                            any_nonpersistent = True
-                            break
-
-                if any_nonpersistent:
-                    time.sleep(5)
+
+        while self._oremote("wait", completions):
+
+            any_nonpersistent = False
+            for c in completions:
```

Comment:
what about changing this to 
```python
any_nonpersistent = any((c.is_read and not c.is_compete) or (not c.is_read and not c.is_persistent) for c in completions)
# afaict same as
any_nonpersistent = any(not (not c.is_read or c.is_compete) or not (c.is_read or c.is_persistent) for c in completions)
# afaict same as
any_nonpersistent = any(not ((not c.is_read or c.is_compete) and (c.is_read or c.is_persistent)) for c in completions)
```
?

Reply:
Brilliant idea!! I have added the "should_wait" attribute to the completion objects. It simplifies and make more easy to understand the behavior in the "wait" calls/workflow.